docs/datamaps/index.md

```diff
@@ -0,0 +1,169 @@
+
+## Syncing
+A synced data map will have its values synced to clients. A data map can be marked as synced using `DataMapType$Builder#synced(Codec<T> networkCodec, boolean mandatory)`.   
+The values of the data map will then be synced using the `networkCodec`.  
+If the `mandatory` flag is set to `true`, clients that do not support the data map (including Vanilla clients) will not be able to connect to the server, nor vice-versa. A non-mandatory data map is, on the other side, optional, so it will not prevent any clients from joining.
+
+## JSON Structure and location
+Data maps are loaded from a JSON file located at `:mapNamespace/data_maps/:registryNamespace/:registryPath/:mapPath.json`.  
+For more information, please [check out the dedicated page](./structure.md).
+
+## Usage
+As data maps can be used on any registry, they can be queried through `Holder`s, and not through the actual registry objects.  
+You can query a data map value using `Holder#getData(DataMapType)`. If that object doesn't have a value attached, the method will return `null`.
+
+:::note
+Only reference holders will return a value in that method. `Direct` holders will **not**.  Generally, you will only encounter reference holders (which are returned by methods such as `Registry#wrapAsHolder`, `Registry#getHolder` or the different `builtInRegistryHolder` methods).
+:::
+
+To continue the example above, we can implement our intended behaviour as follows:
+```java
+public static void onItemDrop(final ItemTossEvent event) {
+    final ItemStack stack = event.getEntity().getItem();
+    // ItemStack has a getItemHolder method that will return a Holder<Item> which points to the item the stack is of
+    //highlight-next-line
+    final DropHealing value = stack.getItemHolder().getData(DROP_HEALING);
+    // Since getData returns null if the item will not have a drop healing value attached, we guard against it being null
+    if (value != null) {
+        // And here we simply use the values
+        if (event.getPlayer().level().getRandom().nextFloat() > value.chance()) {
+            event.getPlayer().heal(value.amount());
+        }
+    }
+}
+```
+
+## Advanced data maps
+Advanced data maps are data maps which have added functionality. Namely, the ability of merging values and selectively removing them, through a remover. Implementing some form of merging and removers is highly recommended for data maps whose values are collection-likes (like `Map`s or `List`s).
+
+`AdvancedDataMapType` have one more generic besides `T` and `R`: `VR extends DataMapValueRemover<R, T>`. This additional generic allows you to datagen remove objects with increased type safety.
+
+### Creation
+You create an `AdvancedDataMapType` using `AdvancedDataMapType#builder`. Unlike the normal builder, the builder returned by that method will have two more methods (`merger` and `remover`), and it will return an `AdvancedDataMapType`.  
+
+Registration methods remain the same.
```

Comment:
Does this need to be a separate section? It seems like it flows logically from above.